app/actions/instancesManager.js

```diff
@@ -41,10 +41,18 @@ export function startInstance(instanceName) {
   return async (dispatch, getState) => {
     const { auth, settings } = getState();
 
+    const config = JSON.parse(
+      await promisify(fs.readFile)(
+        path.join(PACKS_PATH, instanceName, 'config.json')
+      )
+    );
+
     const command = await launchCommand(
       instanceName,
       auth,
-      settings.java.memory,
+      config.overrideMemory === undefined
```

Comment:
This can be written as
`config.overrideMemory ? config.overrideMemory : settings.java.memory`